### ISSUE TITLE
Observed generation field addition in solr-operator code

### ISSUE THREAD

**Issue by Dhru205** [apache/solr-operator#579]
Hi,
I am running solr operator 0.6.0 and solr version 9.1.0 in kubernetes environment and I want to configure the below changes:
I want to add a status field called observed generation in solr-operator code, this field will help me to check how many times changes were made in solr were successful.
Whenever we are describing the CustomResource deployed for a Solr cluster we should be able to display current status of cluster deployment such as Pending, Running or Failed. I want to see this is under status section while describing  CustomResource. Also I want observed generation field to be increase with every successful solr deployment.
So let me know where exactly I need to add this feature in solr operator code.

**Comment by HoustonPutman** [comment]
> Whenever we are describing the CustomResource deployed for a Solr cluster we should be able to display current status of cluster deployment such as Pending, Running or Failed. I want to see this is under status section while describing CustomResource.

Do the `Replicas`, `Ready Replicas` and `Up to date Replicas` values not provide this information already? It's hard for the solr operator to determine whether a non-ready cloud is "pending" or "failed", so this is would be difficult to implement.

> Also I want observed generation field to be increase with every successful solr deployment.
So let me know where exactly I need to add this feature in solr operator code.

`increase with every successful solr deployment` needs to be more clear here. It could mean any update to the StatefulSet, or it could mean any update to the SolrCloud resource. Also what value does this provide?

The statefulSet.Status has an `observedGeneration` field, is this what you are looking for?

**Comment by Dhru205** [comment]
Hi @HoustonPutman 
I am attaching screenshots of one DB which has already in built feature of observed generation so that you can get the clarity of what exactly is the feature that I want to add in Solr.

![MicrosoftTeams-image (3)](https://github.com/apache/solr-operator/assets/69868710/d6470a1c-46b5-4eda-b8a6-183a149a3cdf)

As you can see in the above screenshot I checked the CR of Cassandra.

![operator](https://github.com/apache/solr-operator/assets/69868710/21e290cc-2970-4755-86c2-2e7bc8bcc3ea)

![MicrosoftTeams-image (4)](https://github.com/apache/solr-operator/assets/69868710/f404b730-64a0-4769-8d3e-9e06af2296b5)

The operator is in ready state and observed generation value is 1 since no changes are made in the configurations.

![MicrosoftTeams-image (5)](https://github.com/apache/solr-operator/assets/69868710/a3c2b4a5-f3a1-4396-aaf8-9ca3aaba2e43)
![MicrosoftTeams-image (6)](https://github.com/apache/solr-operator/assets/69868710/63d11c26-b459-4e59-b0a2-c1d0104432f4)

I  changed the CPU and resources of Cassandra and only after all the pods are up and running then only observed Gen value is changed from 1 to 2.

**Comment by HoustonPutman** [comment]
Yeah, given #560 , we will have a pretty good idea as to when operations start/end. This could certainly be implemented. It looks like it's following a similar paradigm to pod status conditions. Not sure when it will get in, but this is certainly possible to do, and will be easier when the linked issue is resolved.

**Comment by HoustonPutman** [comment]
Not sure what you mean by "this feature", but if you mean the locked cluster operations, its a feature of the solr operator and has been merged in for the next release (v0.8.0).

**Comment by rashiga** [comment]
Hi ,
   I could understand that we are showing status per solr Node. Is there any way by which we can show an aggregated status through Observed Generation which tells us that the change in the spec that we have done is applied to the solr pods and they are running successfully?
